### PR TITLE
Dropdown toggle

### LINKED ISSUE #17: [BUG] DropDown toggle
**Describe the bug**
Right now, when we click on any part of the screen and if the dropdown is expanded already, it doesn't close.

**To Reproduce**
Add a dropdown and try expanding and closing by tapping on different areas on screen

**Expected behavior**
DropDown should collapse as soon as the user hits anywhere on the screen(If user hits on the dropdown view itself, it is handled)

**Smartphone (please complete the following information):**
 - Device: iPhone 11 Pro Max simulator
 - OS: 13.3
 - Xcode: 11.3.1

### PR DESCRIPTION
**Pull request by FranklinSamboni**
# Description

This branch fix the bug DropDown toggle #17, The way followed was add a tap gesture to the current top view that allow us to identify when an user tap on any position on the top view. Besides, I added a UIGestureRecognizerDelegate to close de DropDown even if  two gestures are recognized simultaneously (when user focus a UITextField or another type of gesture)

Fixes #17 

## Type of change
- [✓] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?
Please describe on what device it is tested:

Device: iPhone 11 Pro Max simulator
OS: 14.1
Xcode: 12.0.1

# Checklist:

- [✓] My code follows the style guidelines of this project
- [✓] I have performed a self-review of my own code
- [✓] I have commented my code, particularly in hard-to-understand areas 
- [✓] I have made corresponding changes to the documentation
- [✓] My changes generate no new warnings

# Reviewers:
@tejuamirthi

### CODE REVIEW

File: CustomDropDown/View/CustomDropDown.swift [hunk 1]
```diff
@@ -231,38 +232,77 @@ class CustomDropDownView<T>: UIView, UITableViewDataSource, UITableViewDelegate
...
+            dropDown.center.y -= dropDown.frame.height/2
+            dropDown.layoutIfNeeded()
+        })
+    }
+    
+    private func addOutsideGesture() {
+        guard outsideGesture == nil else { return }
+        let topView = window?.topViewController()?.view
+        outsideGesture = UITapGestureRecognizer(target: self, action: #selector(outsideGestureHandler))
+        outsideGesture?.delegate = self
```
Comment (tejuamirthi) [inline CustomDropDown/View/CustomDropDown.swift:282]: you can force unwrap here @FranklinSamboni 
outsideGesture!.delegate = self

**Review by tejuamirthi — CHANGES_REQUESTED** [review]
The selected row is not being updated now(I think the didselect of dropdown tableview is overriden by the new tapgesture). Please check and update. Thanks!

**Comment by tejuamirthi** [comment]
Hey @FranklinSamboni 
you might need to take a pull from main and resolve conflicts! Please update your branch

**Comment by tejuamirthi** [comment]
@FranklinSamboni 
Can you provide a recording of the same? I need to see the selected row to be updated! Thanks

**Comment by FranklinSamboni** [comment]
@tejuamirthi, I want apologize for the delay, I didn't see your comments. I did the necessary changes, now you can test again.

File: CustomDropDown.xcodeproj/xcshareddata/xcschemes/Example.xcscheme [hunk 2]
```diff
@@ -0,0 +1,78 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (tejuamirthi) [inline CustomDropDown.xcodeproj/xcshareddata/xcschemes/Example.xcscheme:1]: @FranklinSamboni 
Do we need this file? If not can you add the pattern in .gitignore, may be I missed? Let me know if this is necessary

**Comment by tejuamirthi** [comment]
> @tejuamirthi, I want apologize for the delay, I didn't see your comments. I did the necessary changes, now you can test again.

@FranklinSamboni Please check and let me know about the new comment. I'll review and merge as soon as possible. Thanks

File: CustomDropDown.xcodeproj/xcshareddata/xcschemes/Example.xcscheme (hunk 2 above)
Comment (FranklinSamboni) [reply CustomDropDown.xcodeproj/xcshareddata/xcschemes/Example.xcscheme:1]: No, It's not necessary, I am going to delete.

Comment (tejuamirthi) [reply CustomDropDown.xcodeproj/xcshareddata/xcschemes/Example.xcscheme:1]: Thank you! Push the changes, I'll merge.

**Review by tejuamirthi — APPROVED** [review]
It's great @FranklinSamboni! Thank you so much for your contribution.

If you'd like to contribute more, look at the open issues or you can create a new one if needed.

**Comment by FranklinSamboni** [comment]
@tejuamirthi  Thanks :D. 👍

**Pull request merged**